### DISCUSSION TITLE
Whisper outputs with length longer than the input audio

### DISCUSSION THREAD

**Discussion by qf6101** [openai/whisper#1107] [Q&A]
The transcribed result of this song https://www.bilibili.com/video/BV1Sy4y1m7za/?vd_source=ed7913e2171bc0b0ac66f9ee12b353c6 gives longer length (28s) than the input (22s). Is it a bug of post processing? I used medium-sized and large models, both raise the problem. The following is the output of large model.

{'text': '在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字',
 'segments': [{'id': 0,
   'seek': 0,
   'start': 0.0,
   'end': 2.0,
   'text': '在我眼裡不得全是你',
   'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 1,
   'seek': 0,
   'start': 2.0,
   'end': 4.0,
   'text': '放下你的名字',
   'tokens': [12744, 4438, 18961, 15940, 22381],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 2,
   'seek': 0,
   'start': 4.0,
   'end': 6.0,
   'text': '寫在我的心裡',
   'tokens': [4510, 104, 3581, 14200, 7945, 11066],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 3,
   'seek': 0,
   'start': 6.0,
   'end': 8.0,
   'text': '一一把你好好的珍惜',
   'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 4,
   'seek': 0,
   'start': 8.0,
   'end': 10.0,
   'text': '在我眼裡不得全是你',
   'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 5,
   'seek': 0,
   'start': 10.0,
   'end': 12.0,
   'text': '放下你的名字',
   'tokens': [12744, 4438, 18961, 15940, 22381],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 6,
   'seek': 0,
   'start': 12.0,
   'end': 14.0,
   'text': '寫在我的心裡',
   'tokens': [4510, 104, 3581, 14200, 7945, 11066],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 7,
   'seek': 0,
   'start': 14.0,
   'end': 16.0,
   'text': '一一把你好好的珍惜',
   'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 8,
   'seek': 0,
   'start': 16.0,
   'end': 18.0,
   'text': '在我眼裡不得全是你',
   'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 9,
   'seek': 0,
   'start': 18.0,
   'end': 20.0,
   'text': '放下你的名字',
   'tokens': [12744, 4438, 18961, 15940, 22381],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 10,
   'seek': 0,
   'start': 20.0,
   'end': 22.0,
   'text': '寫在我的心裡',
   'tokens': [4510, 104, 3581, 14200, 7945, 11066],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 11,
   'seek': 0,
   'start': 22.0,
   'end': 24.0,
   'text': '一一把你好好的珍惜',
   'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 12,
   'seek': 0,
   'start': 24.0,
   'end': 26.0,
   'text': '在我眼裡不得全是你',
   'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695},
  {'id': 13,
   'seek': 0,
   'start': 26.0,
   'end': 28.0,
   'text': '放下你的名字',
   'tokens': [12744, 4438, 18961, 15940, 22381],
   'temperature': 0.0,
   'avg_logprob': -0.26745888590812683,
   'compression_ratio': 1.1290322580645162,
   'no_speech_prob': 0.10490067303180695}],
 'language': 'zh'}

**Comment by murasz** [comment]
> Transcribed result of this song https://www.bilibili.com/video/BV1Sy4y1m7za/?vd_source=ed7913e2171bc0b0ac66f9ee12b353c6 gives longer length (28s) than that of the input audio (22s). Is it a bug of post processing? I used medium and large models, both raise the problem. The following is the output of large model.
> 
> {'text': '在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字寫在我的心裡一一把你好好的珍惜在我眼裡不得全是你放下你的名字',
>  'segments': [{'id': 0,
>    'seek': 0,
>    'start': 0.0,
>    'end': 2.0,
>    'text': '在我眼裡不得全是你',
>    'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 1,
>    'seek': 0,
>    'start': 2.0,
>    'end': 4.0,
>    'text': '放下你的名字',
>    'tokens': [12744, 4438, 18961, 15940, 22381],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 2,
>    'seek': 0,
>    'start': 4.0,
>    'end': 6.0,
>    'text': '寫在我的心裡',
>    'tokens': [4510, 104, 3581, 14200, 7945, 11066],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 3,
>    'seek': 0,
>    'start': 6.0,
>    'end': 8.0,
>    'text': '一一把你好好的珍惜',
>    'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 4,
>    'seek': 0,
>    'start': 8.0,
>    'end': 10.0,
>    'text': '在我眼裡不得全是你',
>    'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 5,
>    'seek': 0,
>    'start': 10.0,
>    'end': 12.0,
>    'text': '放下你的名字',
>    'tokens': [12744, 4438, 18961, 15940, 22381],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 6,
>    'seek': 0,
>    'start': 12.0,
>    'end': 14.0,
>    'text': '寫在我的心裡',
>    'tokens': [4510, 104, 3581, 14200, 7945, 11066],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 7,
>    'seek': 0,
>    'start': 14.0,
>    'end': 16.0,
>    'text': '一一把你好好的珍惜',
>    'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 8,
>    'seek': 0,
>    'start': 16.0,
>    'end': 18.0,
>    'text': '在我眼裡不得全是你',
>    'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 9,
>    'seek': 0,
>    'start': 18.0,
>    'end': 20.0,
>    'text': '放下你的名字',
>    'tokens': [12744, 4438, 18961, 15940, 22381],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 10,
>    'seek': 0,
>    'start': 20.0,
>    'end': 22.0,
>    'text': '寫在我的心裡',
>    'tokens': [4510, 104, 3581, 14200, 7945, 11066],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 11,
>    'seek': 0,
>    'start': 22.0,
>    'end': 24.0,
>    'text': '一一把你好好的珍惜',
>    'tokens': [2257, 2257, 16075, 26410, 20715, 8434, 235, 48199],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 12,
>    'seek': 0,
>    'start': 24.0,
>    'end': 26.0,
>    'text': '在我眼裡不得全是你',
>    'tokens': [3581, 1654, 25281, 11066, 1960, 5916, 11319, 1541, 2166],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695},
>   {'id': 13,
>    'seek': 0,
>    'start': 26.0,
>    'end': 28.0,
>    'text': '放下你的名字',
>    'tokens': [12744, 4438, 18961, 15940, 22381],
>    'temperature': 0.0,
>    'avg_logprob': -0.26745888590812683,
>    'compression_ratio': 1.1290322580645162,
>    'no_speech_prob': 0.10490067303180695}],
>  'language': 'zh'}
> 

Did you check the accuracy of the transcription in "text" output?

**Reply by qf6101** [reply]
If you listen carefully, it seems the vocal repeats twice very closely nearby (within 0.5s). I think this may cause the problem. Here is the groud truth that I labeled: 想着你亲爱的 在我梦里梦的全是你 刻下你的名字在我的心 一定把你好好的珍惜 想着你亲爱的 在我心里想的全是你 许下心愿一生陪着你 你就是我生命的奇迹. You may notice that the text entwines itself in the transcribed output.

**Reply by murasz** [reply]
> If you listen carefully, it seems the vocal repeats twice very closely nearby (within 0.5s). I think this may cause the problem. Here is the groud truth that I labeled: 想着你亲爱的 在我梦里梦的全是你 刻下你的名字在我的心 一定把你好好的珍惜 想着你亲爱的 在我心里想的全是你 许下心愿一生陪着你 你就是我生命的奇迹. You may notice that the text entwines itself in the transcribed output.

Although I'm not an expert of whisper library, I think if you somehow configure "temperature" and try again, you may have a slightly better chance to get the whole text without extra repeats.

**Reply by qf6101** [reply]
Perhaps, but that doesn't account for the inconsistency in lengths.
